### ISSUE TITLE
PAM Authentication failed   and  200 POST /hub/login?next=%2Fhub%2Fspawn 

### ISSUE THREAD

**Issue by sonalidive786** [jupyterhub/jupyterhub#2734]
hi 

i have install jupyter hub on Rhel 7 with help of community site ..

after installation i could able to access the jupyter hub url ..but not able to login , with os user even not by root and root equivalent user ..
in jupyterhub.py   i have whitlisted the user name but still not able to login .
how to solve this

**Comment by ryogesh** [comment]
1) What are the versions of OS, Jupyterhub...
2) Enable Debug on the hub, notebook... and check the log msgs

**Comment by playermanny2** [comment]
@ryogesh  also having this issue...

I'm using ubuntu 18.0.4, jupyterhub 1.0.0 and getting debug

```
[I 2019-10-31 20:48:45.555 JupyterHub app:2120] Using Authenticator: jupyterhub.auth.PAMAuthenticator-1.0.0
[I 2019-10-31 20:48:45.555 JupyterHub app:2120] Using Spawner: dockerspawner.dockerspawner.DockerSpawner-0.10.0
[I 2019-10-31 20:48:45.559 JupyterHub app:1257] Loading cookie_secret from /data/jupyterhub_cookie_secret
[D 2019-10-31 20:48:45.559 JupyterHub app:1424] Connecting to db: sqlite:///jupyterhub.sqlite
[D 2019-10-31 20:48:45.570 JupyterHub orm:718] Stamping empty database with alembic revision 4dc2d5a8c53c
[I 2019-10-31 20:48:45.573 alembic.runtime.migration migration:130] Context impl SQLiteImpl.
[I 2019-10-31 20:48:45.573 alembic.runtime.migration migration:137] Will assume non-transactional DDL.
[I 2019-10-31 20:48:45.581 alembic.runtime.migration migration:356] Running stamp_revision  -> 4dc2d5a8c53c
[D 2019-10-31 20:48:45.582 alembic.runtime.migration migration:558] new branch insert 4dc2d5a8c53c
[I 2019-10-31 20:48:45.637 JupyterHub proxy:460] Generating new CONFIGPROXY_AUTH_TOKEN
[W 2019-10-31 20:48:45.638 JupyterHub app:1532] No admin users, admin interface will be unavailable.
[W 2019-10-31 20:48:45.638 JupyterHub app:1534] Add any administrative users to `c.Authenticator.admin_users` in config.
[I 2019-10-31 20:48:45.638 JupyterHub app:1563] Not using whitelist. Any authenticated user will be allowed.
[D 2019-10-31 20:48:45.659 JupyterHub app:1926] Loaded users:

[I 2019-10-31 20:48:45.666 JupyterHub app:2337] Hub API listening on http://jupyterhub:8080/hub/
[W 2019-10-31 20:48:45.666 JupyterHub proxy:642] Running JupyterHub without SSL.  I hope there is SSL termination happening somewhere else...
[I 2019-10-31 20:48:45.666 JupyterHub proxy:645] Starting proxy @ http://:8000
[D 2019-10-31 20:48:45.666 JupyterHub proxy:646] Proxy cmd: ['configurable-http-proxy', '--ip', '', '--port', '8000', '--api-ip', '127.0.0.1', '--api-port', '8001', '--error-target', 'http://jupyterhub:8080/hub/error']
[D 2019-10-31 20:48:45.669 JupyterHub proxy:561] Writing proxy pid file: jupyterhub-proxy.pid
20:48:46.115 - info: [ConfigProxy] Proxying http://*:8000 to (no default)
20:48:46.117 - info: [ConfigProxy] Proxy API at http://127.0.0.1:8001/api/routes
[D 2019-10-31 20:48:46.119 JupyterHub proxy:681] Proxy started and appears to be up
[D 2019-10-31 20:48:46.119 JupyterHub proxy:314] Fetching routes to check
[D 2019-10-31 20:48:46.120 JupyterHub proxy:765] Proxy: Fetching GET http://127.0.0.1:8001/api/routes
[I 2019-10-31 20:48:46.126 JupyterHub proxy:319] Checking routes
[I 2019-10-31 20:48:46.126 JupyterHub proxy:399] Adding default route for Hub: / => http://jupyterhub:8080
[D 2019-10-31 20:48:46.127 JupyterHub proxy:765] Proxy: Fetching POST http://127.0.0.1:8001/api/routes/
[I 2019-10-31 20:48:46.129 JupyterHub app:2422] JupyterHub is now running at http://:8000
[I 2019-10-31 20:48:52.024 JupyterHub log:174] 302 GET /* -> /hub/* (@63.240.191.250) 0.88ms
[W 2019-10-31 20:48:52.366 JupyterHub log:174] 404 GET /hub/* (@63.240.191.250) 22.71ms
[I 2019-10-31 20:49:41.514 JupyterHub log:174] 302 GET /hub/ -> /hub/login (@63.240.191.250) 0.46ms
[I 2019-10-31 20:49:42.668 JupyterHub log:174] 302 GET / -> /hub/ (@63.240.191.250) 0.43ms
[W 2019-10-31 20:49:50.286 JupyterHub log:174] 404 GET /hub/* (@63.240.191.250) 0.77ms
[I 2019-10-31 20:51:48.579 JupyterHub log:174] 302 GET / -> /hub/ (@63.240.191.250) 0.64ms
[I 2019-10-31 20:51:48.630 JupyterHub log:174] 302 GET /hub/ -> /hub/login (@63.240.191.250) 0.44ms
[I 2019-10-31 20:51:48.686 JupyterHub log:174] 200 GET /hub/login (@63.240.191.250) 4.90ms
[I 2019-10-31 20:52:09.533 JupyterHub log:174] 302 GET / -> /hub/ (@63.240.191.250) 0.51ms
[I 2019-10-31 20:52:09.633 JupyterHub log:174] 302 GET /hub/ -> /hub/login (@63.240.191.250) 0.42ms
[I 2019-10-31 20:52:09.733 JupyterHub log:174] 200 GET /hub/login (@63.240.191.250) 0.80ms
[D 2019-10-31 20:52:09.922 JupyterHub log:174] 200 GET /hub/static/css/style.min.css?v=883a1cf654bd9f5cdc5ca7c91efe7d71 (@63.240.191.250) 1.04ms
[D 2019-10-31 20:52:09.929 JupyterHub log:174] 200 GET /hub/static/components/requirejs/require.js?v=f0cc8bbb2fcef87fc194fecbb632fcfa (@63.240.191.250) 0.60ms
[D 2019-10-31 20:52:10.172 JupyterHub log:174] 200 GET /hub/static/components/bootstrap/dist/js/bootstrap.min.js?v=2f34b630ffe30ba2ff2b91e3f3c322a1 (@63.240.191.250) 0.56ms
[D 2019-10-31 20:52:10.436 JupyterHub log:174] 200 GET /hub/logo (@63.240.191.250) 0.55ms
[D 2019-10-31 20:52:10.636 JupyterHub log:174] 200 GET /hub/static/css/style.min.css.map (@63.240.191.250) 0.56ms
[D 2019-10-31 20:52:21.177 JupyterHub base:478] Setting cookie jupyterhub-session-id: {'httponly': True, 'secure': True}
[D 2019-10-31 20:52:21.177 JupyterHub base:482] Setting cookie for almore.cato: jupyterhub-hub-login
[D 2019-10-31 20:52:21.177 JupyterHub base:478] Setting cookie jupyterhub-hub-login: {'httponly': True, 'secure': True, 'path': '/hub/'}
[I 2019-10-31 20:52:21.178 JupyterHub base:663] User logged in: almore.cato
[D 2019-10-31 20:52:21.178 JupyterHub user:240] Creating <class 'dockerspawner.dockerspawner.DockerSpawner'> for almore.cato:
[I 2019-10-31 20:52:21.180 JupyterHub log:174] 302 POST /hub/login?next= -> /hub/spawn (almore.cato@63.240.191.250) 24.55ms
[I 2019-10-31 20:52:21.275 JupyterHub log:174] 302 GET /hub/spawn -> /hub/login?next=%2Fhub%2Fspawn (@63.240.191.250) 0.50ms
[I 2019-10-31 20:52:21.389 JupyterHub log:174] 200 GET /hub/login?next=%2Fhub%2Fspawn (@63.240.191.250) 0.94ms
[D 2019-10-31 20:52:21.516 JupyterHub log:174] 200 GET /hub/static/css/style.min.css?v=883a1cf654bd9f5cdc5ca7c91efe7d71 (@63.240.191.250) 0.67ms
[D 2019-10-31 20:52:21.522 JupyterHub log:174] 200 GET /hub/static/components/requirejs/require.js?v=f0cc8bbb2fcef87fc194fecbb632fcfa (@63.240.191.250) 0.65ms
[D 2019-10-31 20:52:21.523 JupyterHub log:174] 200 GET /hub/static/components/jquery/dist/jquery.min.js?v=220afd743d9e9643852e31a135a9f3ae (@63.240.191.250) 0.63ms
[D 2019-10-31 20:52:21.873 JupyterHub log:174] 200 GET /hub/static/components/bootstrap/dist/js/bootstrap.min.js?v=2f34b630ffe30ba2ff2b91e3f3c322a1 (@63.240.191.250) 0.53ms
[D 2019-10-31 20:52:21.920 JupyterHub log:174] 200 GET /hub/logo (@63.240.191.250) 0.48ms
[D 2019-10-31 20:52:22.197 JupyterHub log:174] 20
```

**Comment by ryogesh** [comment]
Have you enabled SSL? Can you share the config file (redact any sensitive information)

**Comment by playermanny2** [comment]
@ryogesh  Yea, i have SSL enabled via a traefik proxy not on jupyterhub settings itself. I added 

c.JupyterHub.cookie_options = dict(secure=True)
c.HubAuth.cookie_options = dict(secure=True)

in hopes to force jupyter to use secure reading of cookies, as apparently the proxy could be making jupyter think it isn't secure
```
# Configuration file for JupyterHub
import os

c = get_config()

# We rely on environment variables to configure JupyterHub so that we
# avoid having to rebuild the JupyterHub container every time we change a
# configuration parameter.

# Spawn single-user servers as Docker containers
c.JupyterHub.spawner_class = 'dockerspawner.DockerSpawner'

# Spawn containers from this image
c.DockerSpawner.image = os.environ['DOCKER_NOTEBOOK_IMAGE']

# JupyterHub requires a single-user instance of the Notebook server, so we
# default to using the `start-singleuser.sh` script included in the
# jupyter/docker-stacks *-notebook images as the Docker run command when
# spawning containers.  Optionally, you can override the Docker run command
# using the DOCKER_SPAWN_CMD environment variable.
spawn_cmd = os.environ.get('DOCKER_SPAWN_CMD', "start-singleuser.sh")
c.DockerSpawner.extra_create_kwargs.update({ 'command': spawn_cmd })

# Connect containers to this Docker network
network_name = os.environ['DOCKER_NETWORK_NAME']
c.DockerSpawner.use_internal_ip = True
c.DockerSpawner.network_name = network_name
# Pass the network name as argument to spawned containers
c.DockerSpawner.extra_host_config = { 'network_mode': network_name }

# Explicitly set notebook directory because we'll be mounting a host volume to
# it.  Most jupyter/docker-stacks *-notebook images run the Notebook server as
# user `jovyan`, and set the notebook directory to `/home/jovyan/work`.
# We follow the same convention.
notebook_dir = os.environ.get('DOCKER_NOTEBOOK_DIR') or '/home/jovyan/work'
team_volume = os.environ.get('DATA_VOLUME_CONTAINER_TEAM')
team_volume_dir = os.environ.get('DOCKER_NOTEBOOK_TEAM_DIR')
master_repo_volume_dir = os.environ.get('DOCKER_NOTEBOOK_MASTER_REPO_TEAM_DIR')
c.DockerSpawner.notebook_dir = notebook_dir
# Mount the real user's Docker volume on the host to the notebook user's
# notebook directory in the container
c.DockerSpawner.volumes = { 'jupyterhub-user-{username}': notebook_dir, 'jupyterhub-team-data': team_volume_dir, 'jupyterhub-master-repo': master_repo_volume_dir }

# Remove containers once they are stopped
c.DockerSpawner.remove_containers = True

# For debugging arguments passed to spawned containers
c.DockerSpawner.debug = True

# User containers will access hub by container name on the Docker network
c.JupyterHub.hub_ip = 'jupyterhub'
c.JupyterHub.hub_port = 8080

# Persist hub data on volume mounted inside container
data_dir = os.environ.get('DATA_VOLUME_CONTAINER', '/data')

c.JupyterHub.cookie_secret_file = os.path.join(data_dir,
    'jupyterhub_cookie_secret')


# Set the log level by value or name.
c.JupyterHub.log_level = 'DEBUG'

# Enable debug-logging of the single-user server
c.Spawner.debug = True


c.JupyterHub.cookie_options = dict(secure=True)
c.HubAuth.cookie_options = dict(secure=True)


```

**Comment by ryogesh** [comment]
What is the error on the browser and on the log? Looks like the log is truncated.
[D 2019-10-31 20:52:22.197 JupyterHub log:174] 20

Also, verify the network configuration when using containers,
https://jupyterhub.readthedocs.io/en/stable/getting-started/networking-basics.html#configure-the-hub-if-the-proxy-or-spawners-are-remote-or-isolated

**Comment by playermanny2** [comment]
@ryogesh  hm i'm not seeing any browser error, and I can confirm the network config. Everything works fine if i don't have the secure cookies response header set. Bc of this Jupyter seems to not be able to decrypt the cookie being set to tell the spawner my username to spawn up my user's server.

```

jupyterhub       | [W 2019-11-01 16:53:13.504 JupyterHub configurable:168] Config option `cookie_options` not recognized by `JupyterHub`.
jupyterhub       | [I 2019-11-01 16:53:13.504 JupyterHub app:2120] Using Authenticator: jupyterhub.auth.PAMAuthenticator-1.0.0
jupyterhub       | [I 2019-11-01 16:53:13.505 JupyterHub app:2120] Using Spawner: dockerspawner.dockerspawner.DockerSpawner-0.10.0
jupyterhub       | [I 2019-11-01 16:53:13.508 JupyterHub app:1257] Loading cookie_secret from /data/jupyterhub_cookie_secret
jupyterhub       | [D 2019-11-01 16:53:13.508 JupyterHub app:1424] Connecting to db: sqlite:///jupyterhub.sqlite
jupyterhub       | [D 2019-11-01 16:53:13.519 JupyterHub orm:718] Stamping empty database with alembic revision 4dc2d5a8c53c
jupyterhub       | [I 2019-11-01 16:53:13.523 alembic.runtime.migration migration:130] Context impl SQLiteImpl.
jupyterhub       | [I 2019-11-01 16:53:13.523 alembic.runtime.migration migration:137] Will assume non-transactional DDL.
jupyterhub       | [I 2019-11-01 16:53:13.530 alembic.runtime.migration migration:356] Running stamp_revision  -> 4dc2d5a8c53c
jupyterhub       | [D 2019-11-01 16:53:13.530 alembic.runtime.migration migration:558] new branch insert 4dc2d5a8c53c
jupyterhub       | [I 2019-11-01 16:53:13.733 JupyterHub proxy:460] Generating new CONFIGPROXY_AUTH_TOKEN
jupyterhub       | [W 2019-11-01 16:53:13.734 JupyterHub app:1532] No admin users, admin interface will be unavailable.
jupyterhub       | [W 2019-11-01 16:53:13.734 JupyterHub app:1534] Add any administrative users to `c.Authenticator.admin_users` in config.
jupyterhub       | [I 2019-11-01 16:53:13.734 JupyterHub app:1563] Not using whitelist. Any authenticated user will be allowed.
jupyterhub       | [D 2019-11-01 16:53:13.755 JupyterHub app:1926] Loaded users:
jupyterhub       |
jupyterhub       | [I 2019-11-01 16:53:13.761 JupyterHub app:2337] Hub API listening on http://jupyterhub:8080/hub/
jupyterhub       | [W 2019-11-01 16:53:13.762 JupyterHub proxy:642] Running JupyterHub without SSL.  I hope there is SSL termination happening somewhere else...
jupyterhub       | [I 2019-11-01 16:53:13.762 JupyterHub proxy:645] Starting proxy @ http://:8000
jupyterhub       | [D 2019-11-01 16:53:13.762 JupyterHub proxy:646] Proxy cmd: ['configurable-http-proxy', '--ip', '', '--port', '8000', '--api-ip', '127.0.0.1', '--api-port', '8001', '--error-target', 'http://jupyterhub:8080/hub/error']
jupyterhub       | [D 2019-11-01 16:53:13.765 JupyterHub proxy:561] Writing proxy pid file: jupyterhub-proxy.pid
jupyterhub       | 16:53:14.211 - info: [ConfigProxy] Proxying http://*:8000 to (no default)
jupyterhub       | 16:53:14.212 - info: [ConfigProxy] Proxy API at http://127.0.0.1:8001/api/routes
jupyterhub       | [D 2019-11-01 16:53:14.320 JupyterHub proxy:681] Proxy started and appears to be up
jupyterhub       | [D 2019-11-01 16:53:14.320 JupyterHub proxy:314] Fetching routes to check
jupyterhub       | [D 2019-11-01 16:53:14.321 JupyterHub proxy:765] Proxy: Fetching GET http://127.0.0.1:8001/api/routes
jupyterhub       | [I 2019-11-01 16:53:14.327 JupyterHub proxy:319] Checking routes
jupyterhub       | [I 2019-11-01 16:53:14.327 JupyterHub proxy:399] Adding default route for Hub: / => http://jupyterhub:8080
jupyterhub       | [D 2019-11-01 16:53:14.328 JupyterHub proxy:765] Proxy: Fetching POST http://127.0.0.1:8001/api/routes/
jupyterhub       | [I 2019-11-01 16:53:14.330 JupyterHub app:2422] JupyterHub is now running at http://:8000
jupyterhub       | [D 2019-11-01 16:58:14.332 JupyterHub proxy:765] Proxy: Fetching GET http://127.0.0.1:8001/api/routes
jupyterhub       | [I 2019-11-01 16:58:14.333 JupyterHub proxy:319] Checking routes
jupyterhub       | [D 2019-11-01 17:03:14.330 JupyterHub proxy:765] Proxy: Fetching GET http://127.0.0.1:8001/api/routes
jupyterhub       | [I 2019-11-01 17:03:14.331 JupyterHub proxy:319] Checking routes
jupyterhub       | [I 2019-11-01 17:06:14.613 JupyterHub log:174] 302 GET / -> /hub/ (@63.240.191.250) 1.06ms
jupyterhub       | [I 2019-11-01 17:06:14.714 JupyterHub log:174] 302 GET /hub/ -> /hub/login (@63.240.191.250) 0.56ms
jupyterhub       | [I 2019-11-01 17:06:14.834 JupyterHub log:174] 200 GET /hub/login (@63.240.191.250) 22.09ms
jupyterhub       | [D 2019-11-01 17:06:15.012 JupyterHub log:174] 200 GET /hub/static/css/style.min.css?v=883a1cf654bd9f5cdc5ca7c91efe7d71 (@63.240.191.250) 0.96ms
jupyterhub       | [D 2019-11-01 17:06:15.238 JupyterHub log:174] 200 GET /hub/static/components/requirejs/require.js?v=f0cc8bbb2fcef87fc194fecbb632fcfa (@63.240.191.250) 0.59ms
jupyterhub       | [D 2019-11-01 17:06:15.248 JupyterHub log:174] 200 GET /hub/static/components/jquery/dist/jquery.min.js?v=220afd743d9e9643852e31a135a9f3ae (@63.240.191.250) 0.56ms
jupyterhub       | [D 2019-11-01 17:06:15.345 JupyterHub log:174] 200 GET /hub/static/components/bootstrap/dist/js/bootstrap.min.js?v=2f34b630ffe30ba2ff2b91e3f3c322a1 (@63.240.191.250) 0.53ms
jupyterhub       | [D 2019-11-01 17:06:15.770 JupyterHub log:174] 200 GET /hub/logo (@63.240.191.250) 0.54ms
jupyterhub       | [D 2019-11-01 17:06:32.380 JupyterHub log:174] 200 GET /hub/static/css/style.min.css?v=883a1cf654bd9f5cdc5ca7c91efe7d71 (@63.240.191.250) 0.74ms
jupyterhub       | [D 2019-11-01 17:06:33.143 JupyterHub log:174] 200 GET /hub/static/css/style.min.css.map (@63.240.191.250) 0.58ms
jupyterhub       | [D 2019-11-01 17:07:02.624 JupyterHub log:174] 200 GET /hub/static/css/style.min.css.map (@63.240.191.250) 0.76ms
jupyterhub       | [I 2019-11-01 17:07:07.669 JupyterHub log:174] 302 GET / -> /hub/ (@63.240.191.250) 0.60ms
jupyterhub       | [I 2019-11-01 17:07:07.776 JupyterHub log:174] 302 GET /hub/ -> /hub/login (@63.240.191.250) 0.47ms
jupyterhub       | [I 2019-11-01 17:07:07.889 JupyterHub log:174] 200 GET /hub/login (@63.240.191.250) 0.85ms
jupyterhub       | [D 2019-11-01 17:07:08.125 JupyterHub log:174] 200 GET /hub/static/css/style.min.css?v=883a1cf654bd9f5cdc5ca7c91efe7d71 (@63.240.191.250) 0.67ms
jupyterhub       | [D 2019-11-01 17:07:08.133 JupyterHub log:174] 200 GET /hub/static/components/requirejs/require.js?v=f0cc8bbb2fcef87fc194fecbb632fcfa (@63.240.191.250) 0.59ms
jupyterhub       | [D 2019-11-01 17:07:08.141 JupyterHub log:174] 200 GET /hub/static/components/bootstrap/dist/js/bootstrap.min.js?v=2f34b630ffe30ba2ff2b91e3f3c322a1 (@63.240.191.250) 0.85ms
jupyterhub       | [D 2019-11-01 17:07:08.141 JupyterHub log:174] 200 GET /hub/static/components/jquery/dist/jquery.min.js?v=220afd743d9e9643852e31a135a9f3ae (@63.240.191.250) 1.06ms
jupyterhub       | [D 2019-11-01 17:07:08.381 JupyterHub log:174] 200 GET /hub/logo (@63.240.191.250) 0.49ms
jupyterhub       | [D 2019-11-01 17:07:08.665 JupyterHub log:174] 200 GET /hub/static/css/style.min.css.map (@63.240.191.250) 0.52ms
jupyterhub       | [I 2019-11-01 17:08:13.277 JupyterHub log:174] 302 GET / -> /hub/ (@63.240.191.250) 0.60ms
jupyterhub       | [D 2019-11-01 17:08:14.331 JupyterHub proxy:765] Proxy: Fetching GET http://127.0.0.1:8001/api/routes
jupyterhub       | [I 2019-11-01 17:08:14.333 JupyterHub proxy:319] Checking routes
jupyterhub       | [D 2019-11-01 17:08:58.649 JupyterHub base:478] Setting cookie jupyterhub-session-id: {'httponly': True, 'secure': True}
jupyterhub       | [D 2019-11-01 17:08:58.649 JupyterHub base:482] Setting cookie for almore.cato: jupyterhub-hub-login
jupyterhub       | [D 2019-11-01 17:08:58.649 JupyterHub base:478] Setting cookie jupyterhub-hub-login: {'httponly': True, 'secure': True, 'path': '/hub/'}
jupyterhub       | [I 2019-11-01 17:08:58.649 JupyterHub base:663] User logged in: almore.cato
jupyterhub       | [D 2019-11-01 17:08:58.649 JupyterHub user:240] Creating <class 'dockerspawner.dockerspawner.DockerSpawner'> for almore.cato:
jupyterhub       | [I 2019-11-01 17:08:58.652 JupyterHub log:174] 302 POST /hub/login?next= -> /hub/spawn (almore.cato@63.240.191.250) 25.33ms
jupyterhub       | [I 2019-11-01 17:08:58.768 JupyterHub log:174] 302 GET /hub/spawn -> /hub/login?next=%2Fhub%2Fspawn (@63.240.191.250) 0.57ms
jupyterhub       | [I 2019-11-01 17:08:58.884 JupyterHub log:174] 200 GET /hub/login?next=%2Fhub%2Fspawn (@63.240.191.250) 0.90ms
jupyterhub       | [D 2019-11-01 17:08:59.136 JupyterHub log:174] 200 GET /hub/static/css/style.min.css?v=883a1cf654bd9f5cdc5ca7c91efe7d71 (@63.240.191.250) 0.71ms
jupyterhub       | [D 2019-11-01 17:08:59.139 JupyterHub log:174] 200 GET /hub/static/components/requirejs/require.js?v=f0cc8bbb2fcef87fc194fecbb632fcfa (@63.240.191.250) 0.68ms
jupyterhub       | [D 2019-11-01 17:08:59.140 JupyterHub log:174] 200 GET /hub/static/components/jquery/dist/jquery.min.js?v=220afd743d9e9643852e31a135a9f3ae (@63.240.191.250) 0.60ms
jupyterhub       | [D 2019-11-01 17:08:59.147 JupyterHub log:174] 200 GET /hub/logo (@63.240.191.250) 0.56ms
jupyterhub       | [D 2019-11-01 17:08:59.148 JupyterHub log:174] 200 GET /hub/static/components/bootstrap/dist/js/bootstrap.min.js?v=2f34b630ffe30ba2ff2b91e3f3c322a1 (@63.240.191.250) 0.55ms
jupyterhub       | [D 2019-11-01 17:09:01.377 JupyterHub log:174] 200 GET /hub/static/css/style.min.css.map (@63.240.191.250) 0.56ms
```

**Comment by playermanny2** [comment]
are the cookies supposed to look like this when using set cookies secure and httponly header? with no name
<img width="354" alt="cookies" src="https://user-images.githubusercontent.com/16566894/68045899-d4aed680-fc97-11e9-9e4f-c964853d044a.PNG">

**Comment by playermanny2** [comment]
@ryogesh hey any ideas here? still trying to find a remedy

**Comment by playermanny2** [comment]
This was due to traefik set-cookie being executed before any cookies were actually set. What this meant is that it would override any further cookies with the set-cookie value set. 

In this case, what needed to happen was to edit any response cookies with the added "HttpOnly" and "Secure" Flag. It seemed as if there was no way to handel this in Traefik, so i switched to using Apache as a reverse proxy. This fixed the issue as I was able to use the apache mod_headers directive to do a
```Header edit Set-Cookie ^(.*)$ $1;HttpOnly;Secure```

Thanks for the help
